tagmanager/src/tm_workspace.c

```diff
@@ -489,199 +466,198 @@ static gchar *create_temp_file(const gchar *tpl)
 
 #ifdef HAVE_GLOB_H
 	globbuf.gl_offs = 0;
 
 	if (includes[0][0] == '"')	/* leading \" char for glob matching */
-	for(idx_inc = 0; idx_inc < includes_count; idx_inc++)
 	{
-		size_t dirty_len = strlen(includes[idx_inc]);
-		char *clean_path = g_malloc(dirty_len - 1);
+		for (i = 0; i < includes_count; i++)
+		{
+			size_t dirty_len = strlen(includes[i]);
+			gchar *clean_path;
 
-		strncpy(clean_path, includes[idx_inc] + 1, dirty_len - 1);
-		clean_path[dirty_len - 2] = 0;
+			if (dirty_len < 2)
+				continue;
+
+			clean_path = g_malloc(dirty_len - 1);
+
+			strncpy(clean_path, includes[i] + 1, dirty_len - 1);
+			clean_path[dirty_len - 2] = 0;
 
 #ifdef TM_DEBUG
-		g_message ("[o][%s]\n", clean_path);
+			g_message ("[o][%s]\n", clean_path);
 #endif
-		glob(clean_path, 0, NULL, &globbuf);
+			glob(clean_path, 0, NULL, &globbuf);
 
 #ifdef TM_DEBUG
-		g_message ("matches: %d\n", globbuf.gl_pathc);
+			g_message ("matches: %d\n", globbuf.gl_pathc);
 #endif
 
-		for(idx_glob = 0; idx_glob < globbuf.gl_pathc; idx_glob++)
-		{
+			for (idx_glob = 0; idx_glob < globbuf.gl_pathc; idx_glob++)
+			{
 #ifdef TM_DEBUG
-			g_message (">>> %s\n", globbuf.gl_pathv[idx_glob]);
+				g_message (">>> %s\n", globbuf.gl_pathv[idx_glob]);
 #endif
-			if (!g_hash_table_lookup(includes_files_hash,
-									globbuf.gl_pathv[idx_glob]))
-			{
-				char* file_name_copy = strdup(globbuf.gl_pathv[idx_glob]);
-				g_hash_table_insert(includes_files_hash, file_name_copy,
-									file_name_copy);
+				if (!g_hash_table_lookup(table, globbuf.gl_pathv[idx_glob]))
+				{
+					gchar *file_name_copy = g_strdup(globbuf.gl_pathv[idx_glob]);
+
+					g_hash_table_insert(table, file_name_copy, file_name_copy);
 #ifdef TM_DEBUG
-				g_message ("Added ...\n");
+					g_message ("Added ...\n");
 #endif
+				}
 			}
+			globfree(&globbuf);
+			g_free(clean_path);
 		}
-		globfree(&globbuf);
-		g_free(clean_path);
-  	}
-  	else
-#endif
-	/* no glob support or globbing not wanted */
-	for(idx_inc = 0; idx_inc < includes_count; idx_inc++)
+	}
+	else
+#endif /* HAVE_GLOB_H */
 	{
-		if (!g_hash_table_lookup(includes_files_hash,
-									includes[idx_inc]))
+		/* no glob support or globbing not wanted */
+		for (i = 0; i < includes_count; i++)
 		{
-			char* file_name_copy = strdup(includes[idx_inc]);
-			g_hash_table_insert(includes_files_hash, file_name_copy,
-								file_name_copy);
+			if (!g_hash_table_lookup(table, includes[i]))
+			{
+				gchar* file_name_copy = g_strdup(includes[i]);
+
+				g_hash_table_insert(table, file_name_copy, file_name_copy);
+			}
 		}
-  	}
+	}
 
-	/* Checks for duplicate file entries which would case trouble */
-	g_hash_table_foreach(includes_files_hash, tm_move_entries_to_g_list,
-						 &includes_files);
+	g_hash_table_foreach(table, tm_move_entries_to_g_list, &includes_files);
+	g_hash_table_destroy(table);
 
-	includes_files = g_list_reverse (includes_files);
+	return g_list_reverse(includes_files);
```

Comment:
nothing new in the patch set, but reversing the list here is most probably absurd, as the order in the hash table isn't guaranteed, and unlikely to be neither alphabetical nor in order.